notifier/manager.go

```diff
@@ -283,7 +283,9 @@ func (n *Manager) targetUpdateLoop(tsets <-chan map[string][]*targetgroup.Group)
 func (n *Manager) sendOneBatch() {
 	alerts := n.nextBatch()
 
-	if !n.sendAll(alerts...) {
+	if n.sendAll(alerts...) {
+		n.metrics.delivered.Add(float64(len(alerts)))
```

Comment:
I'm afraid this doesn’t account for per-sm relabeling (`ams.cfg.AlertRelabelConfigs`) in `sendAll()`, which can drop alerts. An am might only be interested in a subset of `alerts`.


Because of those filters/relabeling, I don't think that talking about delivery success for the `alerts` batch, here, across all ams makes sense.

for each am there is `prometheus_notifications_sent_total` and `prometheus_notifications_errors_total`, we can get the delivered alerts per am from there I think.